File: src/BreakfastProvider.Api/Models/Responses/ReservationResponse.cs
namespace BreakfastProvider.Api.Models.Responses;

public record ReservationResponse
{
    public int Id { get; init; }
    public string CustomerName { get; init; } = string.Empty;
    public int TableNumber { get; init; }
    public int PartySize { get; init; }
    public DateTime ReservedAt { get; init; }
    public string Status { get; init; } = string.Empty;
    public string? ContactPhone { get; init; }
    public DateTime CreatedAt { get; init; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/response/ReservationResponse.java
package io.lemonlion.breakfast.model.response;

import java.time.Instant;

/** Twin of C# {@code ReservationResponse}. */
public record ReservationResponse(
        int id,
        String customerName,
        int tableNumber,
        int partySize,
        Instant reservedAt,
        String status,
        String contactPhone,
        Instant createdAt) {
}
